Json decoding exception on assistant run steps

### Description

When deserializing `RunStep` (specifically `RunStepDetails`), the following error occurs:
```
Caused by: kotlinx.serialization.json.internal.JsonDecodingException: Polymorphic serializer was not found for class discriminator 'message_creation'
JSON input: .....ge_creation":{"message_id":"msg_dvN6jgl0m7aby9JKJkcEulTg"}}}
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:24)
	at kotlinx.serialization.json.internal.JsonExceptionsKt.JsonDecodingException(JsonExceptions.kt:32)
	at kotlinx.serialization.json.internal.PolymorphicKt.throwSerializerNotFound(Polymorphic.kt:79)
	at kotlinx.serialization.json.internal.PolymorphicKt.decodeSerializableValuePolymorphic(Polymorphic.kt:68)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:81)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:165)
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:533)
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:70)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:165)
	at com.aallam.openai.api.core.PaginatedList$$serializer.deserialize(PaginatedList.kt:6)
```

### Steps to Reproduce

1. Create an Assistant run
2. Try to get the run steps

### Environment

- **openai-kotlin version**: 3.6.0
- **Kotlin version**: 1.8.0
- **OS**: Ubuntu 22

--- a/openai-client/src/commonTest/kotlin/com/aallam/openai/client/TestAssistants.kt
+++ b/openai-client/src/commonTest/kotlin/com/aallam/openai/client/TestAssistants.kt
@@ -11,7 +11,7 @@ import kotlin.test.assertTrue
 class TestAssistants : TestOpenAI() {
 
     @Test
-    fun listAssistants() = test {
+    fun assistants() = test {
         val request = assistantRequest {
             name = "Math Tutor"
             tools = listOf(AssistantTool.CodeInterpreter)

--- a/openai-client/src/commonTest/kotlin/com/aallam/openai/client/TestRuns.kt
+++ b/openai-client/src/commonTest/kotlin/com/aallam/openai/client/TestRuns.kt
@@ -2,10 +2,14 @@ package com.aallam.openai.client
 
 import com.aallam.openai.api.assistant.AssistantTool
 import com.aallam.openai.api.assistant.assistantRequest
+import com.aallam.openai.api.core.PaginatedList
 import com.aallam.openai.api.core.Status
 import com.aallam.openai.api.model.ModelId
 import com.aallam.openai.api.run.RunRequest
+import com.aallam.openai.api.run.RunStep
+import com.aallam.openai.api.run.RunStepDetails
 import com.aallam.openai.api.run.ThreadRunRequest
+import com.aallam.openai.client.internal.JsonLenient
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
@@ -49,5 +53,75 @@ class TestRuns : TestOpenAI() {
 
         val runs = openAI.runSteps(threadId = run.threadId, runId = run.id)
         assertEquals(0, runs.size)
+    }
+
+    @Test
+    fun runSteps() = test {
+        val json = """
+            {
+              "object": "list",
+              "data": [
+                {
+                  "id": "step_dVK6IlRuFv1z2d8GtB13a8Ff",
+                  "object": "thread.run.step",
+                  "created_at": 1700903454,
+                  "run_id": "run_Sc9g8odBSrMHXFajgIRD1vx6",
+                  "assistant_id": "asst_1UUuuAqyn7mctRo1bf95YV4G",
+                  "thread_id": "thread_R6mlKFGLXAz71vb05JsEPeCf",
+                  "type": "message_creation",
+                  "status": "completed",
+                  "cancelled_at": null,
+                  "completed_at": 1700903455,
+                  "expires_at": null,
+                  "failed_at": null,
+                  "last_error": null,
+                  "step_details": {
+                    "type": "message_creation",
+                    "message_creation": {
+                      "message_id": "msg_h9QzSO20zNuonZhXcwNpbnoR"
+                    }
+                  }
+                },
+                {
+                  "id": "step_kDMRs0mJcg2bN7KzrcxorXI2",
+                  "object": "thread.run.step",
+                  "created_at": 1700903443,
+                  "run_id": "run_Sc9g8odBSrMHXFajgIRD1vx6",
+                  "assistant_id": "asst_1UUuuAqyn7mctRo1bf95YV4G",
+                  "thread_id": "thread_R6mlKFGLXAz71vb05JsEPeCf",
+                  "type": "tool_calls",
+                  "status": "completed",
+                  "cancelled_at": null,
+                  "completed_at": 1700903454,
+                  "expires_at": null,
+                  "failed_at": null,
+                  "last_error": null,
+                  "step_details": {
+                    "type": "tool_calls",
+                    "tool_calls": [
+                      {
+                        "id": "call_MeXdlXsLiVDZGmjgtdzDDoKs",
+                        "type": "code_interpreter",
+                        "code_interpreter": {
+                          "input": "from sympy import symbols, Eq, solve\r\n\r\n# Define the variable\r\nx = symbols('x')\r\n\r\n# Define the equation\r\nequation = Eq(3 * x + 11, 14)\r\n\r\n# Solve the equation\r\nsolution = solve(equation, x)\r\nsolution",
+                          "outputs": [
+                            {
+                              "type": "logs",
+                              "logs": "[1]"
+                            }
+                          ]
+                        }
+                      }
+                    ]
+                  }
+                }
+              ],
+              "first_id": "step_dVK6IlRuFv1z2d8GtB13a8Ff",
+              "last_id": "step_kDMRs0mJcg2bN7KzrcxorXI2",
+              "has_more": false
+            }
+        """.trimIndent()
+        val decoded = JsonLenient.decodeFromString<PaginatedList<RunStep>>(json)
+        println(decoded)
     }
 }

--- a/openai-core/src/commonMain/kotlin/com.aallam.openai.api/run/RunStep.kt
+++ b/openai-core/src/commonMain/kotlin/com.aallam.openai.api/run/RunStep.kt
@@ -8,6 +8,7 @@ import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
 @BetaOpenAI
+@Serializable
 public sealed interface RunStep {
     /**
      * The identifier of the run step, which can be referenced in API endpoints.

--- a/openai-core/src/commonMain/kotlin/com.aallam.openai.api/run/RunStepDetails.kt
+++ b/openai-core/src/commonMain/kotlin/com.aallam.openai.api/run/RunStepDetails.kt
@@ -2,6 +2,7 @@ package com.aallam.openai.api.run
 
 import com.aallam.openai.api.BetaOpenAI
 import com.aallam.openai.api.file.FileId
+import com.aallam.openai.api.message.MessageId
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 
@@ -9,6 +10,7 @@ import kotlinx.serialization.Serializable
  * A run step object.
  */
 @BetaOpenAI
+@Serializable
 public sealed interface RunStepDetails
 
 /**
@@ -32,7 +34,7 @@ public data class MessageCreation(
     /**
      * The ID of the message that was created by this run step.
      */
-    @SerialName("message") public val message: String,
+    @SerialName("message_id") public val messageId: MessageId,
 )
 
 /**
@@ -133,7 +135,7 @@ public sealed interface CodeInterpreterToolCallOutput {
         /**
          * The text output from the Code Interpreter tool call.
          */
-        @SerialName("text") public val text: String,
+        @SerialName("text") public val text: String? = null,
     ) : CodeInterpreterToolCallOutput
 
     /**

--- a/sample/jvm/src/main/kotlin/com/aallam/openai/sample/jvm/Assistants.kt
+++ b/sample/jvm/src/main/kotlin/com/aallam/openai/sample/jvm/Assistants.kt
@@ -8,7 +8,10 @@ import com.aallam.openai.api.core.Status
 import com.aallam.openai.api.message.MessageContent
 import com.aallam.openai.api.message.MessageRequest
 import com.aallam.openai.api.model.ModelId
+import com.aallam.openai.api.run.MessageCreationStep
 import com.aallam.openai.api.run.RunRequest
+import com.aallam.openai.api.run.ToolCallStep
+import com.aallam.openai.api.run.ToolCallsStep
 import com.aallam.openai.client.OpenAI
 import kotlinx.coroutines.delay
 
@@ -56,6 +59,10 @@ suspend fun assistants(openAI: OpenAI) {
         delay(1500)
         val retrievedRun = openAI.getRun(threadId = thread.id, runId = run.id)
     } while (retrievedRun.status != Status.Completed)
+
+    // 5.1 Check run steps
+    val runSteps = openAI.runSteps(threadId = run.threadId, runId = run.id)
+    println("\nRun steps: ${runSteps.size}")
 
     // 6. Display the assistant's response
     val assistantMessages = openAI.messages(thread.id)

